backend/src/main/java/ai/verta/modeldb/experimentRun/FutureExperimentRunDAO.java

```diff
@@ -259,8 +261,10 @@ public FutureExperimentRunDAO(Executor executor, FutureJdbi jdbi, UAC uac) {
                 .execute());
   }
 
-  private InternalFuture<Void> checkProjectPermission(
-      String projId, ModelDBActionEnum.ModelDBServiceActions action) {
+  private InternalFuture<Void> checkEntityPermission(
```

Comment:
Creating this generic method is confusing to the reader. For example, when should I use `checkPermission` or `checkEntityPermission`? This will lead to bugs by using the wrong method.